### DISCUSSION TITLE
Making pybricks better with VS code

### DISCUSSION THREAD

**Discussion by MrGibbage** [pybricks/support#1127] [Show and tell]
I am using VS code and wrote a task to run my pybricks programs. Feel free to use it if you want.

You will need to edit lines 5 and 12:
"label": "Run on Tim",
and
"Tim",

Both of those refer to the robot name, so replace "Tim" with your robot's name.
There's probably room for improvement in the regex for the problemMatcher. I wish the feedback from the compilers in the terminals gave us more feedback (like column numbers for the error location), but oh well.

```jsonc
{
    "version": "2.0.0",
    "tasks": [
        {
        "label": "Run on Tim",
        "type": "shell",
        "command": "pybricksdev.exe",
        "args": [
            "run",
            "ble",
            "--name",
            "Tim",
            "${file}"
            ],
            "problemMatcher": {
                "owner": "python",
                "fileLocation": [
                    "absolute"
                ],
                "pattern": {
                    "regexp": "^(.*)File(.*)(C:(.*)\\.py)(.*)(line(\\s*))([0-9]+),",
                    "file": 3,
                    "line": 8
                }
            },
            "presentation": {
                "echo": true,
                "reveal": "always",
                "focus": false,
                "panel": "shared",
                "showReuseMessage": false,
                "clear": true,
                "revealProblems": "onProblem"
            }
        }
    ]
}
```

And to really take advantage of the VS code tasks, you will want to make a keyboard shortcut (often called "keybindings") to run the task.
Hit ctrl-shift-P, then go to "Preferences: Open Keyboard Shortcuts (JSON)". Paste this in if you want to use ctrl-alt-l, or feel free to come up with your own combination.

The args line needs to match the task label for it to work. Unless your robot is named Tim, you'll probably want to change the args line here.

```jsonc
[
    {
        "key": "ctrl+alt+l",
        "command": "workbench.action.tasks.runTask",
        "args": "Run on Tim"
    }
]
```

I hope it helps!

**Comment by laurensvalk** [comment]
Thank you!
